### PR TITLE
add events

### PR DESCRIPTION
**Pull request by gpsanant**
adds events on share delegation increases and decreases

### CODE REVIEW

File: src/contracts/interfaces/IDelegationManager.sol [hunk 1]
```diff
@@ -87,10 +87,16 @@ interface IDelegationManager {
      */
     event OperatorMetadataURIUpdated(address indexed operator, string metadataURI);
 
-    // @notice Emitted when @param staker delegates to @param operator.
+    /// @notice Emitted whenever an operator's shares are increased for a given strategy
+    event OperatorSharesIncreased(address indexed operator, address staker, IStrategy strategy, uint256 shares);
+
+    /// @notice Emitted whenever an operator's shares are decreased for a given list of strategies
+    event OperatorSharesDecreased(address indexed operator, address staker, IStrategy[] strategies, uint256[] shares);
```
Comment (bowenli86) [inline src/contracts/interfaces/IDelegationManager.sol:94]: why increase is by one strategy, but decrease is by a list of strategy?

Comment (gpsanant) [reply src/contracts/interfaces/IDelegationManager.sol:94]: gas

File: src/contracts/interfaces/IDelegationManager.sol [hunk 2]
```diff
@@ -87,10 +87,16 @@ interface IDelegationManager {
      */
     event OperatorMetadataURIUpdated(address indexed operator, string metadataURI);
 
-    // @notice Emitted when @param staker delegates to @param operator.
+    /// @notice Emitted whenever an operator's shares are increased for a given strategy
+    event OperatorSharesIncreased(address indexed operator, address staker, IStrategy strategy, uint256 shares);
```
[the remark this answers is not in the document]
Comment (gpsanant) [reply src/contracts/interfaces/IDelegationManager.sol:91]: good callout. Fixed: https://github.com/Layr-Labs/eigenlayer-contracts/pull/167/commits/282e2410dc95b04eb8d9e01094f9c34cf27a1c1a

File: src/contracts/interfaces/IDelegationManager.sol (hunk 1 above)
Comment (bowenli86) [reply src/contracts/interfaces/IDelegationManager.sol:94]: IMHO this is potentially a bad idea from data perspective, better keep APIs uniformed - either all single strategy, or all list of strategy

Comment (ChaoticWalrus) [reply src/contracts/interfaces/IDelegationManager.sol:94]: we could always emit events in a loop when decrementing. I'm OK with that.

**Review by ChaoticWalrus — APPROVED** [review]
LGTM

**Pull request merged**